File: C-Abi-Bright-Aot/Tests/C-Abi-Bridge.Aot.Vb.Tests/TestCrypto/TestCryptoChaCha20Poly1305/CryptoChaCha20Poly1305Test.vb
Option Strict On
Option Explicit On

Namespace michele.natale.Tests
  Partial Friend NotInheritable Class CryptoChaCha20Poly1305Test
    Public Shared Sub StartNative(rounds As Int32)
      TestChaCha20Poly1305File(rounds)
      TestChaCha20Poly1305Bytes(rounds)
      TestChaCha20Poly1305BytesStress()
      Console.WriteLine()
    End Sub
  End Class
End Namespace

File: C-Abi-Bridge-Aot/Tests/CAbiBridgeAot.Tests/TestCrypto/TestCryptoCryption/TestCryptoSymmetric/TestCryptoChaCha20Poly1305/CryptoChaCha20Poly1305Test.cs


namespace michele.natale.Tests;


internal sealed partial class CryptoChaCha20Poly1305Test
{
  public static void StartNative(int rounds)
  {
    TestChaCha20Poly1305File(rounds);

    TestChaCha20Poly1305Bytes(rounds);

    TestChaCha20Poly1305BytesStress();

    Console.WriteLine();
  }


  public static void StartManaged(int rounds)
  {

    //Nur für Testzwecken der Managed-Methoden,
    //da mit 'UnmanagedCallersOnly' der Zugriff
    //zu Testszwecken nicht gewährt wird.
    //Bitte für die Nativen Tests dekommentieren!!

    //Only for testing purposes of managed-methods,
    //as access for testing purposes is not granted
    //with ‘UnmanagedCallersOnly’.
    //Please uncomment for native tests!

    TestChaCha20Poly1305FileManaged(rounds);

    TestChaCha20Poly1305BytesManaged(rounds);

    TestChaCha20Poly1305BytesStressManaged();

    Console.WriteLine();
  }
}
